design/dedicated-osd-pod.md

```diff
@@ -0,0 +1,73 @@
+# Run OSD in its own Pod
+
+## TL;DR
+
+A one-OSD-per-Pod placement should be implemented to improve reliability and resource efficiency for Ceph OSD daemon.
+
+## Background
+
+Currently in Rook 0.7, Rook Operator starts a ReplicaSet for Rook OSD  on each storage node. The ReplicaSet has just one replica. Rook OSD scans and prepares devices, creates OSD IDs and data directories, generates Ceph configuration. At last, Rook OSD starts all Ceph OSD daemons in foreground and tracks the OSD processes.
+
+As observed, all Ceph OSDs are running in the same Pod.
+
+## Limitations
+
+One Pod for all OSDs doesn't have the highest reliability nor efficiency. If the Pod is deleted, accidentally or during maintenence, all OSDs are down till the ReplicaSet restart. If nodes have asymmetric configurations, some have more devices than others, the Rook Operator doesn't request appropriate resources for the OSD Pods, since the resources are requested before the OSDs are discovered.
+
+A more comprehensive discussions can be found at [this issue](https://github.com/rook/rook/issues/1341).
+
+## Proposal   
+
+We propose the following OSD startup change.
+
+### Create new OSDs
+| Sequence |Rook Operator  | Rook OSD   | Ceph OSD  | Note  |
+|---|---|---|---|---|
+| 1  | Read devices from cluster CRD and create a Pod for Rook OSD for each storage node  |   |   |  |
+| 2  |   | prepare OSDs  |   | |
+| 3  |   |  Persist OSD ID, datapath, and node info in a per node Configmap | | |
+| 4  | Watch Configmap, parse Configmap, extract OSD info, construct OSD Pod command and arg | | |
+| 5  | Create one deployment per OSD | | |
+| 6  | | | Start OSD Daemon | Running `ceph-osd` in a dedicated Pod eliminates the RBAC rules to work with `configmaps` assigned to Rook OSD  |
+
+### OSD Addition/Deletion
+
+Rook OSD updates Configmap to reflect current OSD devices on the node.
+Rook Operator creates/deletes OSD deployments based on Configmap and Rook CRD change. 
+
+### OSD Pod Naming
+
+Rook Operator creates OSD Pod names using cluster name, node name, and OSD ID.
+
+### Impact
+
+#### Rook Operator
+
+No privilege change, all the RBAC rules assigned to Rook Operator still works.
+
+Rook Operator will watch Configmaps created by OSD s and reconcile Ceph OSD Daemon deployments with the latest information in Configmaps and CRD.
+
+#### Rook OSD 
+
+There is no privilege change, Rook OSD  still uses the same RBAC rules.
+
+However, Rook OSD  no longer exec Ceph OSD daemon. It either continues watching and polling the devices or exits immediately after creating the OSD Configmap.
+
+#### Ceph OSD Daemon
+
+`ceph-osd` is no longer exec'ed by Rook OSD, it becomes the Pod entrypoint and does not need any RBAC rules assigned to OSD . 
```

Comment:
In this section it would be good to list the advantages of running the daemon now. No need for the osd pod to have RBAC privileges (no need to update configmap), no need to run privileged, and no process management.